var/spack/repos/builtin/packages/amdfftw/package.py

```diff
@@ -26,41 +26,151 @@ class Amdfftw(FftwBase):
+        msg='amd-fast-planner is supported from 3.0 onwards')
     conflicts(
         '+amd-fast-planner',
         when='precision=quad',
-        msg="amd-fast-planner doesn't support quad precision")
+        msg='amd-fast-planner is not supported for quad precision')
     conflicts(
         '+amd-fast-planner',
         when='precision=long_double',
-        msg="amd-fast-planner doesn't support long_double precision")
+        msg='amd-fast-planner is not supported for long_double precision')
+    conflicts(
+        '+amd-top-n-planner',
+        when='@:3.0.0',
+        msg='amd-top-n-planner is supported from 3.0.1 onwards')
+    conflicts(
+        '+amd-top-n-planner',
+        when='precision=long_double',
+        msg='amd-top-n-planner is not supported for long_double precision')
+    conflicts(
+        '+amd-top-n-planner',
+        when='precision=quad',
+        msg='amd-top-n-planner is not supported for quad precision')
+    conflicts(
+        '+amd-top-n-planner',
+        when='+amd-fast-planner',
+        msg='amd-top-n-planner cannot be used with amd-fast-planner')
+    conflicts(
+        '+amd-top-n-planner',
+        when='+threads',
+        msg='amd-top-n-planner works only for single thread')
+    conflicts(
+        '+amd-top-n-planner',
+        when='+mpi',
+        msg='amd-top-n-planner is not supported for mpi thread')
+    conflicts(
+        '+amd-top-n-planner',
+        when='+openmp',
+        msg='amd-top-n-planner is not supported for openmp thread')
+    conflicts(
+        '+amd-mpi-vader-limit',
+        when='@:3.0.0',
+        msg='amd-mpi-vader-limit is supported from 3.0.1 onwards')
+    conflicts(
+        '+amd-mpi-vader-limit',
+        when='precision=quad',
+        msg='amd-mpi-vader-limit is not supported for quad precision')
+    conflicts(
+        '+amd-trans',
+        when='+threads',
+        msg='amd-trans works only for single thread')
+    conflicts(
+        '+amd-trans',
+        when='+mpi',
+        msg='amd-trans is not supported for mpi thread')
+    conflicts(
+        '+amd-trans',
+        when='+openmp',
+        msg='amd-trans is not supported for openmp thread')
+    conflicts(
+        '+amd-trans',
+        when='precision=long_double',
+        msg='amd-trans is not supported for long_double precision')
+    conflicts(
+        '+amd-trans',
+        when='precision=quad',
+        msg='amd-trans is not supported for quad precision')
+    conflicts(
+        '+amd-app-opt',
+        when='@:3.0.1',
+        msg='amd-app-opt is supported from 3.1 onwards')
+    conflicts(
+        '+amd-app-opt',
+        when='+mpi',
+        msg='amd-app-opt is not supported for mpi thread')
+    conflicts(
+        '+amd-app-opt',
+        when='precision=long_double',
+        msg='amd-app-opt is not supported for long_double precision')
+    conflicts(
+        '+amd-app-opt',
+        when='precision=quad',
+        msg='amd-trans is not supported for quad precision')
```

Comment:
That whole thing could be simplified. Ex.

```
    when('+amd-top-n-planner'):
        conflicts('+threads', msg='amd-top-n-planner does not support" +threads')
        ...
```

But maybe someone else has even better ideas? Since the `msg` are mostly repetitive you could also simply loop over the conflict condition strings.

Reply:
This change would call for major code refactoring. Need to revalidate

Can I takeup this change in coming release ?

During next release, would follow, following reference spack recipe (which has above suggested change):
https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/quantum-espresso/package.py